src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
+        dphys_dref[1][2] =  0.0;
+        double xtemp = x_ref[0] + alpha*std::sin(pi*x_ref[0])*std::sin(pi*x_ref[1]); 
+        double ytemp = x_ref[1] + alpha*exp(1.0-x_ref[1])*std::sin(pi*x_ref[0])*std::sin(pi*x_ref[1]); 
+        dphys_dref[2][0] = 1.0/10.0*pi*(std::cos(2.0*pi*xtemp)*dphys_dref[0][0] + std::cos(2.0*pi*ytemp)*dphys_dref[1][0]);
+        dphys_dref[2][1] = 1.0/10.0*pi*(std::cos(2.0*pi*xtemp)*dphys_dref[0][1] + std::cos(2.0*pi*ytemp)*dphys_dref[1][1]);
+        dphys_dref[2][2] = 1.0;
+    }
+
+    return dphys_dref;
+}
+
+template<int dim>
+std::unique_ptr<dealii::Manifold<dim,dim> > CurvManifold<dim>::clone() const 
+{
+    return std::make_unique<CurvManifold<dim>>();
+}
+
+template <int dim, int nstate>
+dealii::Point<dim> AdvectionPeriodic<dim,nstate>
+::warp (const dealii::Point<dim> &p)
+{
+    const double pi = atan(1)*4.0;
+    dealii::Point<dim> q = p;
+
+    const double beta = 1.0/10.0;
+    const double alpha = 1.0/10.0;
+    if (dim == 1){
+        q[dim-1] = p[dim-1] + cos(2.0 * pi* p[dim-1]);
+    }
+    if (dim == 2){
+        //non sym transform
+        q[dim-2] =p[dim-2] +  beta*std::cos(pi/2.0 * p[dim-2]) * std::cos(3.0 * pi/2.0 * p[dim-1]);
+        q[dim-1] =p[dim-1] +  beta*std::sin(2.0 * pi * (p[dim-2])) * std::cos(pi /2.0 * p[dim-1]);
+
+        //Jesse Chan non sym transform
+
+//        q[0] =p[0] +beta* std::sin(pi*4.0 * (p[0])) * std::sin(pi * (p[1]+1));
+//        q[1] =p[1] -beta* 0.5*std::sin(pi*2.0 * (p[0])) * std::sin(pi /2.0 * (p[1]+1));
+
+        //Gassner nonsym
+
+//        q[0] =p[0] - 0.1*std::sin(2.0*pi*p[1]); 
+//        q[1] =p[1] + 0.1*std::sin(2.0*pi*p[0]);
+
+    }
+    if(dim==3){
+       //periodic non sym wild
+        double temp1 = alpha*(std::sin(pi * p[0]) * std::sin(pi * p[1]));
+        double temp2 = alpha*exp(1.0-p[1])*(std::sin(pi * p[0]) * std::sin(pi* p[1]));
+        q[0] =p[0] + temp1; 
+        q[1] =p[1] + temp2;
+        q[2] =p[2] +  1.0/20.0*( std::sin(2.0 * pi * q[0]) + std::sin(2.0 * pi * q[1]));
+
+    }
+
+    return q;
+}
+/****************************
+ * End of Curvilinear Grid
+ * ***************************/
+
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_energy(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg) const
+{
+	double energy = 0.0;
+        dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->right_hand_side);
+        dg->global_mass_matrix.vmult( Mu_hat, dg->solution);
+        energy = dg->solution * Mu_hat;
+	return sqrt(energy);
+}
+
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_conservation(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg, const double poly_degree) const
+{
+	double conservation = 0.0;
+       // dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->solution.size());
+        dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->right_hand_side);
```

Comment:
Avoid using symbols as variable names like Mu_hat unless you put a reference as to where exactly this Mu_hat is defined.

Reply:
fixed